flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartitionTest.java

```diff
@@ -179,10 +185,49 @@ public void testFlushWithUnfinishedBufferBehindFinished() throws Exception {
 		try {
 			subpartition.add(createFilledBufferConsumer(1025)); // finished
 			subpartition.add(createFilledBufferBuilder(1024).createBufferConsumer()); // not finished
+			long oldNumNotifications = availablityListener.getNumNotifications();
 			subpartition.flush();
+			// buffer queue is > 1, should already be notified, no further notification necessary
+			assertThat(oldNumNotifications, greaterThan(0L));
+			assertEquals(oldNumNotifications, availablityListener.getNumNotifications());
 
 			assertNextBuffer(readView, 1025, true, 1, false, true);
 			assertNextBuffer(readView, 1024, false, 1, false, false);
+			assertNoNextBuffer(readView);
+		} finally {
+			subpartition.release();
+		}
+	}
+
+	/**
+	 * A flush call with a buffer size of 1 should always notify consumers (unless already flushed).
+	 */
+	@Test
+	public void testFlushWithUnfinishedBufferBehindFinished2() throws Exception {
+		final ResultSubpartition subpartition = createSubpartition();
```

Comment:
Extract those initialisations to setup/teardown/`@Rule`. This code block is duplicated couple of times:

```
		final ResultSubpartition subpartition = createSubpartition();
		AwaitableBufferAvailablityListener availablityListener = new AwaitableBufferAvailablityListener();
		ResultSubpartitionView readView = subpartition.createReadView(availablityListener);
		availablityListener.resetNotificationCounters();

		(...)
		} finally {
				subpartition.release();
		}
```

Reply:
unfortunately, not every unit test works on the same setup - are you proposing to
- instantiate these nonetheless and let those be unused in some tests, or
- split the unit test into one with and one without this initialization?
Or maybe I'm not aware of some trick that solves this...

Reply:
ok, I forked off the methods with read view and availability listener to `PipelinedSubpartitionWithReadViewTest`